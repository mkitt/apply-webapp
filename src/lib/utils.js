// @flow
import { __, addIndex, always, and, complement, either, gt, isEmpty, isNil, map } from 'ramda'
import { withProps } from 'recompose'

// -------------------------------------
// Environments
export const isDev = always(process.env.NODE_ENV === 'development')

// -------------------------------------
// Types
export const isNotNil = complement(isNil)
export const isNotEmpty = complement(isEmpty)
export const isEmptyOrNil = either(isNil, isEmpty)
export const isNotEmptyAndNotNil = and(isNotNil, isNotEmpty)

// -------------------------------------
// Relation
export const gt0 = gt(__, 0)

// -------------------------------------
// Lists
export const nool = always([])()
export const mapIndexed = addIndex(map)

// -------------------------------------
// Objects
export const noob = always({})()

// -------------------------------------
// Functions
export const noop = () => {}

export const log = withProps(props => (
  // eslint-disable-next-line no-console
  console.log(props.displayName || 'üëç', props) && props
))

export const warn = withProps(props => (
  // eslint-disable-next-line no-console
  console.warn(props.displayName || '‚ö†Ô∏è', props) && props
))
