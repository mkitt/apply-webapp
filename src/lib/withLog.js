import { withProps } from 'recompose'

export default withProps(props => (
  // eslint-disable-next-line
  console.log(props.logName || props.displayName || '👍', props) && props
))
