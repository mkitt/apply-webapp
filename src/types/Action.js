// @flow
import type { Applicant } from './Applicant'

export type Action = {
  applicants: Array<Applicant>,
  id: ID,
  label: string,
  message: string,
  name: string,
}
