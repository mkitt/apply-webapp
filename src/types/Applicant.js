// @flow
import { camelize } from 'humps'
import { compose, prop, split, head, last } from 'ramda'
import type { Action } from './Action'
import type { Application } from './Application'

export default undefined

export type Applicant = {
  action: ?Action,
  applications: Array<Application>,
  birthDate: ?string,
  createdAt: DateTime,
  email: string,
  github: string,
  id: ID,
  name: string,
  referredBy: ?string,
  resume: ?string,
  startDate: ?DateTime,
  status: ?string,
  updatedAt: DateTime,
  // derived data...
  firstName: string,
  lastName: string,
}

// -------------------------------------
// Derived helpers

export const deriveFirstName = compose(
  head,
  split(' '),
  prop('name'),
)

export const deriveLastName = compose(
  last,
  split(' '),
  prop('name'),
)

// -------------------------------------
// For testing content
export const stub = (props: Applicant) => {
  const firstName = (props != null && props.firstName) || 'Kenny'
  const lastName = (props != null && props.lastName) || 'Bania'
  const camel = camelize(`${firstName} ${lastName}`)
  const applicant = {
    id: camel,
    action: null,
    applications: [],
    birthDate: '06/06/2006',
    createdAt: '2010-10-31T12:12:12.000Z',
    email: 'kenny.bania@example.com',
    firstName: 'Kenny',
    github: camel,
    lastName,
    referredBy: 'Simon Taranto',
    resume: 'https://www.resume.com',
    startDate: null,
    status: 'pending',
    updatedAt: '2010-12-24T24:24:24.000Z',
  }
  return {
    ...applicant,
    ...props,
  }
}
