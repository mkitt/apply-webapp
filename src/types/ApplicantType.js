// @flow
export type ApplicantType = {
  applyAction: string,
  applyStatus: string,
  birthdate: string,
  createdAt: string,
  email: string,
  firstName: string,
  github: string,
  id: string,
  lastName: string,
  referredBy: string,
  resume: string,
  scoreLogicEvaluation: number,
  scoreOnlineLogicTest: number,
  scoreValuesEvaluation: number,
  startDate: string,
  updatedAt: string,
}
