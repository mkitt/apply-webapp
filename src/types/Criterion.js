// @flow
import { camelize } from 'humps'
import { compose, prop } from 'ramda'
import type { Evaluation } from './Evaluation'

export type Criterion = {
  id: ID,
  createdAt: DateTime,
  evaluation: ?Evaluation,
  title: string,
  options: Array<string>,
  questions: Array<string>,
  updatedAt: DateTime,
  // derived...
  name: string,
}

// -------------------------------------
// Derived helpers

export const deriveName = compose(
  camelize,
  prop('title'),
)

// -------------------------------------
// For testing content
export const stub = (props: Criterion) => {
  const title = (props != null && props.title) || 'Problem Solving'
  const camel = camelize(title)
  return {
    id: camel,
    title,
    name: camel,
    options: [
      `${title} Option 3`,
      `${title} Option 2`,
      `${title} Option 1`,
    ],
    questions: [
      `${title} Question 1`,
      `${title} Question 2`,
      `${title} Question 3`,
    ],
    ...props,
  }
}

export default undefined
