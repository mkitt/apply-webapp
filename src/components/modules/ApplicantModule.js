// @flow
import React from 'react'
import glamorous from 'glamorous'
import TimeAgo from 'react-timeago'
import type { Match } from 'react-router-dom'
import { map } from 'ramda'
import { isNotNil } from '../../lib/utils'
import { COLORS } from '../../constants/theme'
import ConfirmDialog from '../modules/ConfirmDialog'
import SendStatusDialog from '../modules/SendStatusDialog'
import DescriptionBase from '../blocks/Description'
import EvaluationSummary from '../blocks/EvaluationSummary'
import Loader from '../blocks/Loader'
import Modal from '../blocks/Modal'
import Status from '../blocks/Status'
import Button from '../elements/Button'
import Heading from '../elements/Heading'
import SectionContainment from '../elements/SectionContainment'
import StatusBar from '../elements/StatusBar'
import TextLink from '../elements/TextLink'
import type { Application } from '../../types/Application'

type Props = {
  application: Application,
  evaluatorList?: string,
  handleCloseModal?: () => {},
  handleOpenSendStatus?: () => void,
  handleOpenSubmitEvaluation?: () => {},
  handleStatusMutation: () => void,
  handleStatusSelect: () => {},
  isModalOpen?: boolean,
  isSendStatusDialog?: boolean,
  isSubmitEvaluationDialog?: boolean,
  match: Match,
  showScores?: boolean,
  statusValue: string,
}

const ApplicantModule = ({
  application: {
    id,
    applicant,
    evaluations,
    isStatusMutatable,
    resume,
    // scoreLogicEvaluation,
    scoreOnlineLogicTest,
    // scoreValuesEvaluation,
    status,
    statusLabel,
  },
  evaluatorList,
  handleCloseModal,
  handleOpenSendStatus,
  handleOpenSubmitEvaluation,
  handleStatusMutation,
  handleStatusSelect,
  isModalOpen,
  isSendStatusDialog,
  isSubmitEvaluationDialog,
  match,
  showScores,
  statusValue,
}: Props) => (
  <div>
    { isNotNil(status) &&
      <StatusBar />
    }
    <SectionContainment>
      { isNotNil(status) &&
        <Header>
          <Heading>
            {`${applicant.firstName} ${applicant.lastName}`}
          </Heading>
          { isNotNil(statusLabel) &&
            <Status
              id={id}
              align="flex-end"
              onClick={isStatusMutatable ? handleStatusMutation : null}
            >
              {statusLabel}
            </Status>
          }
        </Header>
      }
      { isNotNil(applicant.email) &&
        <Description term="Email Address">
          <TextLink href={`mailto:${applicant.email}`}>{applicant.email}</TextLink>
        </Description>
      }
      { isNotNil(applicant.birthDate) &&
      <Description term="Birth Date">
        {applicant.birthDate}
      </Description>
      }
      { isNotNil(applicant.login) &&
        <Description term="Login">
          { applicant.loginLink != null ?
            <TextLink href={applicant.loginLink}>{applicant.login}</TextLink> :
            <span>{applicant.login}</span>
          }
        </Description>
      }
      { isNotNil(applicant.referredBy) &&
        <Description term="Referred By">
          {applicant.referredBy}
        </Description>
      }
      { isNotNil(applicant.createdAt) &&
        <Description term="Application Age">
          <TimeAgo date={applicant.createdAt} />
        </Description>
      }
      { isNotNil(evaluatorList) &&
        <Description term="Evaluators">
          {evaluatorList}
        </Description>
      }
      { isNotNil(resume) &&
        <Description term="Resume">
          <TextLink href={`${String(resume)}`}>View PDF</TextLink>
        </Description>
      }
    </SectionContainment>
    { showScores === true &&
      <SectionContainment mt={72}>
        <EvaluationSummary
          score={scoreOnlineLogicTest}
          term="Online Logic Test"
        />
        { map(evaluation => (
          <EvaluationSummary
            key={evaluation.id}
            term={evaluation.title}
            to={`${match.url}/evaluation/${evaluation.id}`}
          />
        ), evaluations)}
      </SectionContainment>
    }
    {(isNotNil(handleOpenSendStatus) || isNotNil(handleOpenSubmitEvaluation)) &&
      <Actions>
        { isNotNil(handleOpenSendStatus) && isNotNil(isSendStatusDialog) &&
          <Button box centered primary mt={32} onClick={handleOpenSendStatus}>
            Send status email
          </Button>
        }
        { isNotNil(handleOpenSubmitEvaluation) && isNotNil(isSubmitEvaluationDialog) &&
          <Button box centered primary mt={32} onClick={handleOpenSubmitEvaluation}>
            Submit Evaluation
          </Button>
        }
      </Actions>
    }
    { isModalOpen != null && handleCloseModal != null &&
      <Modal
        isOpen={isModalOpen}
        onRequestClose={handleCloseModal}
      >
        { isSendStatusDialog === true &&
          <SendStatusDialog
            handleCancel={handleCloseModal}
            handleChange={handleStatusSelect}
            handleConfirm={handleCloseModal}
            statusValue={statusValue}
          />
        }
        { isSubmitEvaluationDialog === true &&
          <ConfirmDialog
            handleCancel={handleCloseModal}
            handleConfirm={handleCloseModal}
          />
        }
      </Modal>
    }
  </div>
)

ApplicantModule.defaultProps = {
  evaluatorList: null,
  handleCloseModal: null,
  handleOpenSendStatus: null,
  handleOpenSubmitEvaluation: null,
  isModalOpen: null,
  isSendStatusDialog: false,
  isSubmitEvaluationDialog: false,
  showScores: false,
}

export default ApplicantModule
export const ComponentLoader = () => (
  <Loader>Loading Applicant...</Loader>
)

// -------------------------------------

const Description = props => (
  <DescriptionBase color={COLORS.GREY_8} isRow {...props} />
)

const Header = glamorous.header({
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center',
  minHeight: 128,
})

const Actions = glamorous(SectionContainment)({
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'flex-end',
})
