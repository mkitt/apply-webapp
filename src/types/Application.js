// @flow
import { compose, contains, prop, replace } from 'ramda'
import { capitalize } from '../lib/utils'
import type { Applicant } from './Applicant'
import type { Evaluator } from './Evaluator'

export type Application = {
  id: ID,
  applicant: Applicant,
  complete: boolean,
  evaluators: ?Array<Evaluator>,
  resume: ?string,
  scoreLogicEvaluation: ?Int,
  scoreOnlineLogicTest: ?Int,
  scoreValuesEvaluation: ?Int,
  status: string,
  // derived...
  isStatusMutatable: boolean,
  statusLabel: string,
}

// -------------------------------------
// Derived helpers

export const deriveIsStatusMutatable = compose(
  contains('needs'),
  prop('status'),
)

export const deriveStatusLabel = compose(
  capitalize,
  replace(/_/gi, ' '),
  prop('status'),
)

// -------------------------------------
// For testing content
export const stub = (props: Application) => {
  const application = {
    id: '1',
    complete: false,
    evaluators: null,
    isStatusMutatable: false,
    resume: 'uploads/asset/application-resume.pdf',
    scoreLogicEvaluation: 20,
    scoreOnlineLogicTest: 40,
    scoreValuesEvaluation: 20,
    status: 'pending',
    statusLabel: 'Pending',
  }
  return {
    ...application,
    ...props,
  }
}

export default undefined
