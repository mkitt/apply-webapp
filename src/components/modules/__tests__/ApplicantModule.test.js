import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import ApplicantModule from '../ApplicantModule'
import { noop } from '../../../lib/utils'
import { stub } from '../../../types/Applicant'
import { stub as stubAction } from '../../../types/Action'
import { stub as stubApplication } from '../../../types/Application'

const match = {
  url: '/match/url',
}

describe('<ApplicantModule />', () => {
  it('shows the correct snapshot tree for the <ApplicantModule> module', () => {
    const component = (
      <MemoryRouter>
        <ApplicantModule
          applicant={stub()}
          match={match}
        />
      </MemoryRouter>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot tree for the <ApplicantModule> module with data', () => {
    const applicant = stub({
      action: stubAction(),
      applications: [stubApplication()],
    })

    const component = (
      <MemoryRouter>
        <ApplicantModule
          applicant={applicant}
          evaluatorList="Evaluator 1, Evaluator 2"
          match={match}
          handleCloseModal={noop}
          handleOpenConfirm={noop}
          handleOpenSendRecommendation={noop}
          handleOpenSendStatus={noop}
          handleRecommendationSelect={noop}
          handleStatusSelect={noop}
          isConfirmDialog
          recommendationValue="Recommendation Value"
          statusValue="status value"
        />
      </MemoryRouter>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
