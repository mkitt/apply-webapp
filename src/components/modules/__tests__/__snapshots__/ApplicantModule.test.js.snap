// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ApplicantModule /> shows the correct snapshot tree for the <ApplicantModule> module 1`] = `
.css-0,
[data-css-0] {
  height: 4px;
  background-color: #4d4d4d;
}

.css-1,
[data-css-1] {
  position: relative;
  max-width: 960px;
  margin-right: auto;
  margin-left: auto;
  padding-right: 16px;
  padding-left: 16px;
}

.css-2,
[data-css-2] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: space-between;
  align-items: center;
  min-height: 128px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-3,
[data-css-3] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  font-size: 34px;
  font-weight: 200;
  color: #808080;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

.css-4,
[data-css-4] {
  position: relative;
  z-index: 2;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  margin-top: 8px;
  margin-bottom: 8px;
  align-items: flex-end;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
  -webkit-box-align: end;
  -webkit-align-items: flex-end;
}

.css-5,
[data-css-5] {
  font-size: 14px;
  font-weight: 600;
  text-transform: uppercase;
  color: #999;
}

.css-6,
[data-css-6] {
  margin-top: 8px;
  margin-left: 0;
}

.css-7,
[data-css-7] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  align-items: center;
  margin-top: 8px;
  font-size: 14px;
  font-weight: 400;
  text-transform: uppercase;
  color: #4d4d4d;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-8,
[data-css-8] {
  margin-right: 8px;
}

.css-9,
[data-css-9] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  margin-top: 8px;
  margin-bottom: 8px;
  color: #808080;
  flex-flow: row nowrap;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
  -webkit-flex-flow: row nowrap;
}

.css-10,
[data-css-10] {
  font-size: 14px;
  font-weight: 600;
  text-transform: uppercase;
  color: currentColor;
  min-width: 192px;
}

.css-11,
[data-css-11] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  align-items: center;
  margin-left: 0;
  font-size: 16px;
  font-weight: 200;
  color: currentColor;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-12,
[data-css-12] {
  position: relative;
  z-index: 2;
  color: #35c5fc;
  text-decoration: none;
  transition: color 200ms;
  -webkit-transition: color 200ms;
  -moz-transition: color 200ms;
}

.css-13,
[data-css-13] {
  position: relative;
  max-width: 960px;
  margin-right: auto;
  margin-left: auto;
  padding-right: 16px;
  padding-left: 16px;
  margin-top: 72px;
}

.css-14,
[data-css-14] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-top: 44px;
  border-bottom: 1px solid #ccc;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-15,
[data-css-15] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-flow: row nowrap;
  font-size: 28px;
  font-weight: 300;
  color: #808080;
  -webkit-flex-flow: row nowrap;
}

.css-16,
[data-css-16] {
  color: #8ad30e;
  margin-left: 8px;
}

.css-16::before,
[data-css-16]::before {
  content: " : ";
  color: #808080;
}

.css-18,
[data-css-18] {
  position: relative;
  z-index: 2;
  color: #35c5fc;
  text-decoration: none;
  transition: color 200ms;
  font-size: 18px;
  -webkit-transition: color 200ms;
  -moz-transition: color 200ms;
}

<div>
  <div
    className="css-0"
  />
  <section
    className="css-1"
  >
    <header
      className="css-2"
    >
      <h2
        className="css-3"
      >
        Kenny Bania
      </h2>
      <dl
        className="css-4"
      >
        <dt
          className="css-5"
        >
          Application Status
        </dt>
        <dd
          className="css-6"
        >
          <span
            className="css-7"
          >
            <svg
              className="css-8"
              height={32}
              viewBox="0 0 24 24"
              width={32}
            >
              <g
                fill="currentColor"
              >
                <path
                  d="M24,0 C24,0 23.4836048,9.80252996 16.4820958,15.2719707 L16.4128111,19.0346251 L11.4820194,24 L11.4746834,20.2882431 L9.63105705,18.6292449 C8.29671435,22.7691304 1.26439406,22.7691304 1.26439406,22.7691304 C1.26439406,22.7691304 1.26439406,16.3396736 5.58319762,14.4450082 L3.94220876,12.6079505 L-2.84217094e-14,12.6079505 L5.07440302,7.66843342 L8.7285621,7.58627726 C14.4472436,0.671411685 24,0 24,0 Z M16.425,10.1644246 C17.8333261,10.1644246 18.975,9.01830529 18.975,7.60449544 C18.975,6.19068562 17.8333261,5.0445663 16.425,5.0445663 C15.0166739,5.0445663 13.875,6.19068562 13.875,7.60449544 C13.875,9.01830529 15.0166739,10.1644246 16.425,10.1644246 Z"
                />
              </g>
            </svg>
            Pending
          </span>
        </dd>
      </dl>
    </header>
    <dl
      className="css-9"
    >
      <dt
        className="css-10"
      >
        Email Address
      </dt>
      <dd
        className="css-11"
      >
        <a
          className="css-12"
          href="mailto:kenny.bania@example.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          kenny.bania@example.com
        </a>
      </dd>
    </dl>
    <dl
      className="css-9"
    >
      <dt
        className="css-10"
      >
        Birth Date
      </dt>
      <dd
        className="css-11"
      >
        06/06/2006
      </dd>
    </dl>
    <dl
      className="css-9"
    >
      <dt
        className="css-10"
      >
        Login
      </dt>
      <dd
        className="css-11"
      >
        <a
          className="css-12"
          href="https://github.com/kennyBania"
          rel="noopener noreferrer"
          target="_blank"
        >
          kennyBania
        </a>
      </dd>
    </dl>
    <dl
      className="css-9"
    >
      <dt
        className="css-10"
      >
        Referred By
      </dt>
      <dd
        className="css-11"
      >
        Simon Taranto
      </dd>
    </dl>
    <dl
      className="css-9"
    >
      <dt
        className="css-10"
      >
        Application Age
      </dt>
      <dd
        className="css-11"
      >
        <time
          dateTime="2010-10-31T12:12:12.000Z"
          title="2010-10-31T12:12:12.000Z"
        >
          7 years ago
        </time>
      </dd>
    </dl>
    <dl
      className="css-9"
    >
      <dt
        className="css-10"
      >
        Resume
      </dt>
      <dd
        className="css-11"
      >
        <a
          className="css-12"
          href="uploads/asset/application-resume.pdf"
          rel="noopener noreferrer"
          target="_blank"
        >
          View PDF
        </a>
      </dd>
    </dl>
  </section>
  <section
    className="css-13"
  >
    <div
      className="css-14"
    >
      <h2
        className="css-15"
      >
        <span
          className="css-nil"
        >
          Online Logic Test
        </span>
        <span
          className="css-16"
        >
          40
        </span>
      </h2>
    </div>
    <div
      className="css-14"
    >
      <h2
        className="css-15"
      >
        <span
          className="css-nil"
        >
          Logic Evaluation
        </span>
        <span
          className="css-16"
        >
          20
        </span>
      </h2>
      <a
        className="css-18"
        fontSize={18}
        href="/match/url/logic-evaluation"
        onClick={[Function]}
      >
        Re-Score Evaluation
      </a>
    </div>
    <div
      className="css-14"
    >
      <h2
        className="css-15"
      >
        <span
          className="css-nil"
        >
          Values Evaluation
        </span>
        <span
          className="css-16"
        >
          20
        </span>
      </h2>
      <a
        className="css-18"
        fontSize={18}
        href="/match/url/values-evaluation"
        onClick={[Function]}
      >
        Re-Score Evaluation
      </a>
    </div>
  </section>
</div>
`;

exports[`<ApplicantModule /> shows the correct snapshot tree for the <ApplicantModule> module with data 1`] = `
.css-0,
[data-css-0] {
  height: 4px;
  background-color: #4d4d4d;
}

.css-1,
[data-css-1] {
  position: relative;
  max-width: 960px;
  margin-right: auto;
  margin-left: auto;
  padding-right: 16px;
  padding-left: 16px;
}

.css-2,
[data-css-2] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: space-between;
  align-items: center;
  min-height: 128px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-3,
[data-css-3] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  font-size: 34px;
  font-weight: 200;
  color: #808080;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

.css-4,
[data-css-4] {
  position: relative;
  z-index: 2;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  margin-top: 8px;
  margin-bottom: 8px;
  align-items: flex-end;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
  -webkit-box-align: end;
  -webkit-align-items: flex-end;
}

.css-5,
[data-css-5] {
  font-size: 14px;
  font-weight: 600;
  text-transform: uppercase;
  color: #999;
}

.css-6,
[data-css-6] {
  margin-top: 8px;
  margin-left: 0;
}

.css-7,
[data-css-7] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  align-items: center;
  margin-top: 8px;
  font-size: 14px;
  font-weight: 400;
  text-transform: uppercase;
  color: #4d4d4d;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-8,
[data-css-8] {
  margin-right: 8px;
}

.css-9,
[data-css-9] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  margin-top: 8px;
  margin-bottom: 8px;
  color: #808080;
  flex-flow: row nowrap;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
  -webkit-flex-flow: row nowrap;
}

.css-10,
[data-css-10] {
  font-size: 14px;
  font-weight: 600;
  text-transform: uppercase;
  color: currentColor;
  min-width: 192px;
}

.css-11,
[data-css-11] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  align-items: center;
  margin-left: 0;
  font-size: 16px;
  font-weight: 200;
  color: currentColor;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-12,
[data-css-12] {
  position: relative;
  z-index: 2;
  color: #35c5fc;
  text-decoration: none;
  transition: color 200ms;
  -webkit-transition: color 200ms;
  -moz-transition: color 200ms;
}

.css-13,
[data-css-13] {
  position: relative;
  max-width: 960px;
  margin-right: auto;
  margin-left: auto;
  padding-right: 16px;
  padding-left: 16px;
  margin-top: 72px;
}

.css-14,
[data-css-14] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-top: 44px;
  border-bottom: 1px solid #ccc;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-15,
[data-css-15] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-flow: row nowrap;
  font-size: 28px;
  font-weight: 300;
  color: #808080;
  -webkit-flex-flow: row nowrap;
}

.css-16,
[data-css-16] {
  color: #8ad30e;
  margin-left: 8px;
}

.css-16::before,
[data-css-16]::before {
  content: " : ";
  color: #808080;
}

.css-18,
[data-css-18] {
  position: relative;
  z-index: 2;
  color: #35c5fc;
  text-decoration: none;
  transition: color 200ms;
  font-size: 18px;
  -webkit-transition: color 200ms;
  -moz-transition: color 200ms;
}

<div>
  <div
    className="css-0"
  />
  <section
    className="css-1"
  >
    <header
      className="css-2"
    >
      <h2
        className="css-3"
      >
        Kenny Bania
      </h2>
      <dl
        className="css-4"
      >
        <dt
          className="css-5"
        >
          Application Status
        </dt>
        <dd
          className="css-6"
        >
          <span
            className="css-7"
          >
            <svg
              className="css-8"
              height={32}
              viewBox="0 0 24 24"
              width={32}
            >
              <g
                fill="currentColor"
              >
                <path
                  d="M24,0 C24,0 23.4836048,9.80252996 16.4820958,15.2719707 L16.4128111,19.0346251 L11.4820194,24 L11.4746834,20.2882431 L9.63105705,18.6292449 C8.29671435,22.7691304 1.26439406,22.7691304 1.26439406,22.7691304 C1.26439406,22.7691304 1.26439406,16.3396736 5.58319762,14.4450082 L3.94220876,12.6079505 L-2.84217094e-14,12.6079505 L5.07440302,7.66843342 L8.7285621,7.58627726 C14.4472436,0.671411685 24,0 24,0 Z M16.425,10.1644246 C17.8333261,10.1644246 18.975,9.01830529 18.975,7.60449544 C18.975,6.19068562 17.8333261,5.0445663 16.425,5.0445663 C15.0166739,5.0445663 13.875,6.19068562 13.875,7.60449544 C13.875,9.01830529 15.0166739,10.1644246 16.425,10.1644246 Z"
                />
              </g>
            </svg>
            Pending
          </span>
        </dd>
      </dl>
    </header>
    <dl
      className="css-9"
    >
      <dt
        className="css-10"
      >
        Email Address
      </dt>
      <dd
        className="css-11"
      >
        <a
          className="css-12"
          href="mailto:kenny.bania@example.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          kenny.bania@example.com
        </a>
      </dd>
    </dl>
    <dl
      className="css-9"
    >
      <dt
        className="css-10"
      >
        Birth Date
      </dt>
      <dd
        className="css-11"
      >
        06/06/2006
      </dd>
    </dl>
    <dl
      className="css-9"
    >
      <dt
        className="css-10"
      >
        Login
      </dt>
      <dd
        className="css-11"
      >
        <a
          className="css-12"
          href="https://github.com/kennyBania"
          rel="noopener noreferrer"
          target="_blank"
        >
          kennyBania
        </a>
      </dd>
    </dl>
    <dl
      className="css-9"
    >
      <dt
        className="css-10"
      >
        Referred By
      </dt>
      <dd
        className="css-11"
      >
        Simon Taranto
      </dd>
    </dl>
    <dl
      className="css-9"
    >
      <dt
        className="css-10"
      >
        Application Age
      </dt>
      <dd
        className="css-11"
      >
        <time
          dateTime="2010-10-31T12:12:12.000Z"
          title="2010-10-31T12:12:12.000Z"
        >
          7 years ago
        </time>
      </dd>
    </dl>
    <dl
      className="css-9"
    >
      <dt
        className="css-10"
      >
        Evaluators
      </dt>
      <dd
        className="css-11"
      >
        Evaluator 1, Evaluator 2
      </dd>
    </dl>
    <dl
      className="css-9"
    >
      <dt
        className="css-10"
      >
        Resume
      </dt>
      <dd
        className="css-11"
      >
        <a
          className="css-12"
          href="uploads/asset/application-resume.pdf"
          rel="noopener noreferrer"
          target="_blank"
        >
          View PDF
        </a>
      </dd>
    </dl>
  </section>
  <section
    className="css-13"
  >
    <div
      className="css-14"
    >
      <h2
        className="css-15"
      >
        <span
          className="css-nil"
        >
          Online Logic Test
        </span>
        <span
          className="css-16"
        >
          40
        </span>
      </h2>
    </div>
    <div
      className="css-14"
    >
      <h2
        className="css-15"
      >
        <span
          className="css-nil"
        >
          Logic Evaluation
        </span>
        <span
          className="css-16"
        >
          20
        </span>
      </h2>
      <a
        className="css-18"
        fontSize={18}
        href="/match/url/logic-evaluation"
        onClick={[Function]}
      >
        Re-Score Evaluation
      </a>
    </div>
    <div
      className="css-14"
    >
      <h2
        className="css-15"
      >
        <span
          className="css-nil"
        >
          Values Evaluation
        </span>
        <span
          className="css-16"
        >
          20
        </span>
      </h2>
      <a
        className="css-18"
        fontSize={18}
        href="/match/url/values-evaluation"
        onClick={[Function]}
      >
        Re-Score Evaluation
      </a>
    </div>
  </section>
</div>
`;
